src/testdir/test_vim9_import.vim

```diff
@@ -3694,4 +3694,42 @@ def Test_import_member_initializer()
   v9.CheckScriptSuccess(lines)
 enddef
 
+def Test_import_name_conflict_with_local_variable()
+  var lines =<< trim END
+    vim9script
+
+    export class Foo
+      def Method()
+        echo 'method'
+      enddef
+    endclass
+  END
+  writefile(lines, 'Xvim9.vim', 'D')
+
+  lines =<< trim END
+    vim9script
+    import './Xvim9.vim'
+
+    def Main(): void
+      var vim9 = Xvim9.Foo.new()
+      vim9.Method()
+    enddef
+
+    Main()
+  END
+  v9.CheckScriptSuccess(lines)
+
+  lines =<< trim END
+    import './Xvim9.vim'
+
+    function! s:Main() abort
+      let Xvim9 = s:Xvim9.Foo.new()
+      call Xvim9.Method()
```

Comment:
Can you use "assert_equal()" here to check for the returned value?

Reply:
OK. I updated the patch.